mock_drivers/rosbag_mock_drivers/src/USFControllerDriver.cpp

```diff
@@ -0,0 +1,249 @@
+      boost::make_shared<ROSComms<cav_msgs::RobotEnabled>>(CommTypes::pub, false, 10, robot_status_topic_);
+  
+  ////vehicle command topic is subscribed by the vehiclesender node.
+
+  /*vehicle_cmd_ptr_ = boost::make_shared<ConstPtrRefROSComms<autoware_msgs::VehicleCmd>>(
+      std::bind(&MockControllerDriver::vehicleCmdCallback, this, std::placeholders::_1), CommTypes::sub, false, 10,
+      vehicle_cmd_topic_);*/
+
+  enable_robotic_ptr_ =
+      boost::make_shared<ROSComms<cav_srvs::SetEnableRobotic::Request&, cav_srvs::SetEnableRobotic::Response&>>(
+          std::bind(&MockControllerDriver::enableRoboticSrv, this, std::placeholders::_1, std::placeholders::_2),
+          CommTypes::srv, enable_robotic_srv_);
+}
+
+bool MockControllerDriver::onSpin()
+{
+  cav_msgs::RobotEnabled robot_status;
+  robot_status.robot_active = robot_active_;
+  robot_status.robot_enabled = robot_enabled_;
+
+  mock_driver_node_.publishDataNoHeader<cav_msgs::RobotEnabled>(robot_status_topic_, robot_status);
+
+  return true;
+}
+
+unsigned int MockControllerDriver::getRate()
+{
+  return 20;  // 20 Hz as default spin rate to match expected controller data rate
+}
+
+//USF controller message structure
+struct CommandData {
+  double linear_x;
+  double angular_z;
+
+  void reset();
+};
+
+void CommandData::reset()
+{
+  linear_x      = 0;
+  angular_z     = 0;
+}
+
+static CommandData command_data;
+
+static void vehicleCmdCallback(const autoware_msgs::VehicleCmd& msg)
+{
+  command_data.linear_x = msg.twist_cmd.twist.linear.x;
+  command_data.angular_z = msg.twist_cmd.twist.angular.z;
+}
+
+static void *sendCommand(void *arg)
+{
+  int *client_sockp = static_cast<int*>(arg);
+  int client_sock = *client_sockp;
+  delete client_sockp;
+
+  std::ostringstream oss;
+  oss << command_data.linear_x << ",";
+  oss << command_data.angular_z;
+
+  std::string cmd(oss.str());
+  ssize_t n = write(client_sock, cmd.c_str(), cmd.size());
+  if(n < 0){
+    std::perror("write");
+    return nullptr;
+  }
+
+  if(close(client_sock) == -1){
+    std::perror("close");
+    return nullptr;
+  }
+
+  std::cout << "cmd: " << cmd << ", size: " << cmd.size() << std::endl;
+  return nullptr;
+}
+
+static void* receiverCaller(void *unused)
+{
+  constexpr int listen_port = 10001;
+
+  int sock = socket(AF_INET, SOCK_STREAM, 0);
+  if(sock == -1){
+    std::perror("socket");
+    return nullptr;
+  }
+
+  sockaddr_in addr;
+  sockaddr_in client;
+  socklen_t len = sizeof(client);
+
+  std::memset(&addr, 0, sizeof(sockaddr_in));
+  addr.sin_family = PF_INET;
+  addr.sin_port = htons(listen_port);
+  addr.sin_addr.s_addr = INADDR_ANY;
+
+  int ret = bind(sock, (struct sockaddr *)&addr, sizeof(addr));
+  if(ret == -1){
+    std::perror("bind");
+    goto error;
+  }
+
+  ret = listen(sock, 20);
+  if(ret == -1){
+    std::perror("listen");
+    goto error;
+  }
+  while(true){
+    std::cout << "Waiting access..." << std::endl;
+
+    int *client_sock = new int();
```

Comment:
Is there a reason this is initialized as an int* on the stack? It looks like it could be an int just fine without needing to be a pointer, especially considering i don't see it get `delete`'d anywhere.